### PR TITLE
Introduce Support for Multi-CTC Set transcode 

### PR DESCRIPTION
**Pull request by vibhoothi**
This PR Introduces support for Multiple CTC set transcode option and download an XLSM file for the CTC, 

Note that this is not fully tested as I do not have big clusters, so I am staging the PR for review, 

Secondly, there is additional commits from #313 which is required for this MR to be merged as standalone.

Edit:
Screenshots:
image
image
image

In the backend, we are creating a folder each for different video set, and assigning first video-set as 'task' for front-end rendering.

### CODE REVIEW

**Review by vibhoothi — COMMENTED** [review]
Depends on https://github.com/xiph/rd_tool/pull/121

**Comment by vibhoothi** [comment]
[info copy.txt](https://github.com/xiph/awcy/files/9659585/info.copy.txt)

File: awcy_server.ts [hunk 1]
```diff
@@ -406,6 +406,11 @@ app.post('/submit/job',function(req,res) {
   if (!req.body.save_encode) {
     req.body.save_encode = true
   }
+  if (!req.body.ctcSets) {
+    req.body.ctcSets = []
+  } else {
+    req.body.ctcSets = req.body.ctcSets.split(',')
```
Comment (tdaede) [inline awcy_server.ts:412]: Consider using an extended-format urlencoded array https://expressjs.com/en/resources/middleware/body-parser.html#bodyparserurlencodedoptions

**Review by tdaede — APPROVED** [review]

**Comment by tdaede** [comment]
One thing I don't like about the backend having the combined sets (ctc-mandatory and ctc-all) is that a job where each subset is individually selected is going to have a different .json than one where ctc-mandatory is selected, so it will make filtering in the future harder, e.g. if we search for all runs that include A1.

**Pull request merged**